tests/instancetype_test.go

```diff
@@ -655,6 +659,403 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			}, 5*time.Minute, time.Second).Should(Succeed())
 		})
 	})
+
+	FContext("with inferFromVolume enabled", Ordered, func() {
+
+		var (
+			err          error
+			vm           *v1.VirtualMachine
+			instancetype *instancetypev1alpha2.VirtualMachineInstancetype
+			preference   *instancetypev1alpha2.VirtualMachinePreference
+			pvc          *k8sv1.PersistentVolumeClaim
+		)
+
+		const (
+			inferFromVolumeName    = "volume"
+			instancetypeName       = "instancetype"
+			preferenceName         = "preference"
+			pvcSourceName          = "pvc"
+			dataSourceName         = "datasource"
+			dataVolumeTemplateName = "datatemplate"
+		)
+
+		checkVMhasInferredInstancetypeAndPreference := func() {
+			By("Creating and starting the VirtualMachine")
+			vm, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(vm)
+			Expect(err).ToNot(HaveOccurred())
+			vm = tests.StartVMAndExpectRunning(virtClient, vm)
+
+			By("Checking that a VirtualMachineInstance has been created with the original VirtualMachineInstancetype and VirtualMachinePreference applied")
+			Expect(vm.Spec.Instancetype.Name).To(Equal(instancetype.Name))
+			Expect(vm.Spec.Instancetype.Kind).To(Equal(instancetypeapi.SingularResourceName))
+			Expect(vm.Spec.Instancetype.InferFromVolume).To(BeEmpty())
+			Expect(vm.Spec.Preference.Name).To(Equal(preference.Name))
+			Expect(vm.Spec.Preference.Kind).To(Equal(instancetypeapi.SingularPreferenceResourceName))
+			Expect(vm.Spec.Preference.InferFromVolume).To(BeEmpty())
+
+			By("Checking that a VirtualMachineInstance has been created with the inferred VirtualMachineInstancetype and VirtualMachinePreference applied")
+			vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vmi.Spec.Domain.CPU.Cores).To(Equal(instancetype.Spec.CPU.Guest))
+		}
+
+		BeforeAll(func() {
+			if !libstorage.HasCDI() {
+				Skip("instance type and preference inferFromVolume tests require CDI to be installed providing the DataVolume and DataSource CRDs")
+			}
+
+			By("Creating a VirtualMachineInstancetype")
+			instancetype = newVirtualMachineInstancetype(nil)
+			instancetype.Name = instancetypeName
```

Comment:
Nothing to fill into the instancetype's spec like below for the preference?

Reply:
No it already comes with the required CPU and memory resources provided.

Reply:
ACK